core/modules/ccontrol/UserQuery.h

```diff
@@ -80,17 +80,31 @@ class UserQuery {
     // Delegate objects
     virtual std::shared_ptr<qdisp::MessageStore> getMessageStore() = 0;
 
+    /// This method should disappear when we start supporting results
+    /// in locations other than MySQL tables. We'll switch to getResultLocation()
+    /// at that point.
     /// @return Name of the result table for this query, can be empty
-    virtual std::string getResultTableName() = 0;
+    virtual std::string getResultTableName() const { return std::string(); }
+
+    /// Result location could be something like "table:table_name" or
+    /// "file:/path/to/file.csv".
+    /// @return Result location for this query, can be empty
+    virtual std::string getResultLocation() const { return std::string(); }
 
     /// @return ORDER BY part of SELECT statement to be executed by proxy
-    virtual std::string getProxyOrderBy() = 0;
+    virtual std::string getProxyOrderBy() const { return std::string(); }
+
+    /// @return this query's QueryId.
+    virtual QueryId getQueryId() const { return QueryId(0); }
```

Comment:
Same response - this is the only interface that we have now, no other place to stick this method without redesigning the whole thing.